### PR TITLE
Refactor common pod patterns to e2e framework.

### PR DESCRIPTION
**Pull request by timstclair**
Priveledged pods could benifit from this as well, but I'll wait for https://github.com/kubernetes/kubernetes/pull/26228 to go in.

Prerequisite for https://github.com/kubernetes/kubernetes/issues/26215

### CODE REVIEW

**Comment by timstclair** [comment]
Ready for review.

/cc @pwittrock

File: test/e2e/framework/framework.go [hunk 1]
```diff
@@ -93,6 +93,11 @@ type TestDataSummary interface {
 type FrameworkOptions struct {
 	ClientQPS   float32
 	ClientBurst int
+
+	// Tie the framework to a specific node (e.g. for node e2e tests).
+	// NOTE: This must be a pointer because the node-name flag has not been parsed yet when the
+	// framework options are set.
```
Comment (timstclair) [inline test/e2e/framework/framework.go]: To reviewers:
I'm not satisfied with this solution, but don't have a better idea that doesn't require a significant refactoring. If you have any ideas, I'm all ears 👂👂!

Comment (vishh) [inline test/e2e/framework/framework.go]: One of the goals is to be able to share tests between node and cluster level e2es. With this model, one will have to inject the framework into the tests. 
Instead, would it be possible to use `TestContext` to pass the NodeName as a flag and have these tests use a framework method for spawning pods? That way the tests can be re-used as is.

**Comment by vishh** [comment]
The intention of this change is super helpful for sharing tests between test frameworks. If we can move the nodename config to be a flag via `TestContext`, I think PR is good to go.

Sorry for the latency in reviewing this.

**Comment by timstclair** [comment]
@vishh Done.

File: test/e2e/framework/pods.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	"k8s.io/kubernetes/pkg/api"
+
+	. "github.com/onsi/gomega"
+)
+
+// TODO: Consolidate pod-specific framework functions here.
+
+// Create a new pod according to the framework specifications, and wait for it to start.
+func (f *Framework) CreatePod(pod *api.Pod) {
+	f.mungePodSpec(pod)
```
Comment (vishh) [inline test/e2e/framework/pods.go]: nit: Why not refactor this logic into a separate method and share it with `CreatePod` and `CreatePods`?

File: test/e2e_node/container_manager_test.go [hunk 3]
```diff
@@ -50,7 +51,7 @@ var _ = Describe("Kubelet Container Manager", func() {
 					},
 					Spec: api.PodSpec{
 						// Force the Pod to schedule to the node without a scheduler running
-						NodeName: *nodeName,
+						NodeName: framework.TestContext.NodeName,
```
Comment (vishh) [inline test/e2e_node/container_manager_test.go]: nit: Why not use the newly added `framework.CreatePod()` here?

**Comment by vishh** [comment]
A few more nits/comments. Let's get this PR merged quickly.

File: test/e2e/framework/pods.go (hunk 2 above)
Comment (timstclair) [inline test/e2e/framework/pods.go]: I'm not sure exactly what you mean here - are you suggesting calling CreatePod from CreatePods? I wanted to start all the pods in parallel, which is why I didn't do that (the old code would wait for each pod to be running before starting the next)

File: test/e2e_node/container_manager_test.go (hunk 3 above)
Comment (timstclair) [inline test/e2e_node/container_manager_test.go]: I'm not sure why I missed this one. Done.

File: test/e2e/framework/pods.go (hunk 2 above)
Comment (vishh) [inline test/e2e/framework/pods.go]: This is just a minor nit.

```go
func (f *Framework) createPod(pod *api.Pod) error {
 f.mungePodSpec(pod)
 _, err := f.Client.Pods(f.Namespace.Name).Create(pod)
 return err
}

func (f *Framework) CreatePod(pod *api.Pod) {
 ExpectNoError(f.createPod(pod), "Error creating Pod")
 ExpectNoError(f.WaitForPodRunning(pod.Name))
}

func (f *Framework) CreatePods(pods []*api.Pod) {
 for _, pod := range pods {
  ExpectNoError(f.createPod(pod), "Error creating Pod")
 }
 // Wait for pods to be running.
}
```

Comment (timstclair) [inline test/e2e/framework/pods.go]: Done. Naming ok?

Comment (vishh) [inline test/e2e/framework/pods.go]: Did you push your changes?

Comment (timstclair) [inline test/e2e/framework/pods.go]: Yes. Refresh?

**Comment by timstclair** [comment]
Depends on https://github.com/kubernetes/kubernetes/pull/26831 for reusing the framework.

**Comment by vishh** [comment]
LGTM

**Comment by vishh** [comment]
@k8s-bot test node e2e issue #IGNORE

**Comment by goltermann** [comment]
This one needs rebase soon if it's making it into 1.3.

**Comment by goltermann** [comment]
Removing 1.3 milestone - this doesn't appear to be required to ship 1.3.  Please comment if you disagree.

**Comment by vishh** [comment]
@timstclair Ping

**Comment by timstclair** [comment]
I put this on hold since it was removed from 1.3. Are we merging non 1.3 PRs yet?

**Comment by goltermann** [comment]
No, but we plan to open up for post-v1.3 work early next week.

**Comment by vishh** [comment]
HEAD is now open for PRs. Let's get this through!

**Comment by timstclair** [comment]
Rebased & squashed - PTAL

There's a bit more work to do, but I'd like to take this as a starting point before the rebases get out of hand :)

**Comment by timstclair** [comment]
There were 2 flags which I enabled for node e2e's in this PR which are causing the tests to fail:
```
"e2e-verify-service-account", true, "If true tests will verify the service account before running."
"delete-namespace", true, "If true tests will delete namespace after completion. It is only designed to make debugging easier, DO NOT turn it off by default."
```

The first (verify service account) I think does not pertain to the node tests as they stand, so I moved this out of common flags.

The second (delete namespaces) _should_ be enabled for the node tests, but is not currently working. I'd like to disable it for now (to push this PR through) and follow up with a fix to enable it.

@vishh PTAL

**Comment by timstclair** [comment]
@k8s-bot node e2e test this. issue: #IGNORE

**Comment by timstclair** [comment]
Squashed commits, reapplying LGTM.

**Comment by timstclair** [comment]
Fixed rebase errors.

**Comment by timstclair** [comment]
Hmm, this seems to be introducing some flakiness. I'll dig into the failures to see if this is just surfacing existing issues or creating new ones.

**Comment by timstclair** [comment]
@k8s-bot node e2e test this. issue: #28236

**Comment by timstclair** [comment]
@k8s-bot node e2e test this. issue: #IGNORE

**Pull request merged**